src/components/course-aside/course-aside.tsx

```diff
@@ -6,25 +7,43 @@ import CourseAsideProps from './types';
 
 import styles from './styles';
 
+function generateSocialSharingURLs(
+  socialUrl: string,
+  paramConfig: { [k: string]: string | boolean }
+) {
+  const keys = Object.keys(paramConfig);
+  const params = keys.reduce(
+    (acc, val, idx) =>
+      `${acc}${val}=${encodeURIComponent(paramConfig[val])}${
+        idx !== keys.length ? '&' : ''
+      }`,
+    '?'
+  );
+  return `${socialUrl}${params}`;
+}
+
 export default function CourseAside({
   level,
-  length,
+  title,
+  duration,
   openForm,
-  wrapperStyles,
-  prerequisites,
+  wrapperStyles = {},
 }: CourseAsideProps): JSX.Element {
+  const [currentPageUrl, setCurrentPageUrl] = useState<string>('');
+
+  // because of SSR, we need to assure the component mounts before trying to access window obj
+  useEffect(() => {
+    setCurrentPageUrl(window.location.href);
+  }, []);
+
   return (
-    <aside
-      sx={{
-        ...styles.CourseAsideWrapper,
-        ...wrapperStyles,
-      }}
-      data-testid="course-aside"
-    >
+    <aside sx={{ ...wrapperStyles }} data-testid="course-aside">
       <section sx={styles.CourseAsideSection}>
         <div sx={styles.CourseAsideWidget}>
           <h3 sx={styles.CourseAsideLabel}>Length</h3>
-          <span sx={styles.CourseAsideDetail}>{length}</span>
+          <span sx={styles.CourseAsideDetail}>
+            {duration} Hour-long lectures
```

Comment:
I think this should just be `{duration} lectures`. I remember talking about it with someone and these lectures vary in length.

Reply:
Wait actually it should just be `{duration} hours` (or `{duration} hour`), sorry got that mixed up.

Reply:
updating